isolated-jobs-test/src/main/java/com/hazelcast/jet/tests/isolatedjobs/IsolatedJobsStreamTest.java

```diff
@@ -0,0 +1,159 @@
+import static com.hazelcast.jet.tests.common.Util.waitForJobStatus;
+import static com.hazelcast.jet.tests.isolatedjobs.JetMemberSelectorUtil.excludeMember;
+
+/**
+ * This test requires Enterprise License with Advanced Compute Feature.
+ * To run this test locally, you have to use EE version (it can be forced by adding dependency to hazelcast-enterprise
+ * in the pom.xml) and license must be set (it can be done in {@link AbstractSoakTest#run(String[])})
+ */
+public class IsolatedJobsStreamTest extends AbstractSoakTest {
+    private static final int DEFAULT_LOG_VERIFICATION_COUNT_THRESHOLD = 5;
+    private static final int DEFAULT_SLEEP_BETWEEN_VALIDATIONS_IN_MINUTES = 1;
+    private static final int DEFAULT_SNAPSHOT_INTERVAL_MS = 1000;
+    private static final int DEFAULT_SLEEP_BETWEEN_STREAM_RECORDS_MS = 1000 * 15;
+    private int snapshotIntervalMs;
+    private int logVerificationCountThreshold;
+    private int sleepBeforeValidationInMinutes;
+    private int sleepBetweenStreamRecordsMs;
+
+    public static void main(final String[] args) throws Exception {
+        new IsolatedJobsStreamTest().run(args);
+    }
+
+    @Override
+    protected boolean runOnBothClusters() {
+        return true;
+    }
+
+    @Override
+    protected void init(HazelcastInstance client) throws Exception {
+        snapshotIntervalMs = propertyInt("snapshotIntervalMs",
+                DEFAULT_SNAPSHOT_INTERVAL_MS);
+        logVerificationCountThreshold = propertyInt("logVerificationCountThreshold",
+                DEFAULT_LOG_VERIFICATION_COUNT_THRESHOLD);
+        sleepBeforeValidationInMinutes = propertyInt("sleepBeforeValidationInMinutes",
+                DEFAULT_SLEEP_BETWEEN_VALIDATIONS_IN_MINUTES);
+        sleepBetweenStreamRecordsMs = propertyInt("sleepBetweenStreamRecordsMs",
+                DEFAULT_SLEEP_BETWEEN_STREAM_RECORDS_MS);
+    }
+
+    @Override
+    protected void test(HazelcastInstance client, String name) throws Throwable {
+        long begin = System.currentTimeMillis();
+        String logPrefix = "[" + name + "] ";
+
+        Member excludedMember = client.getCluster().getMembers().iterator().next();
+        UUID excludedMemberUuid = excludedMember.getUuid();
+        String excludedMemberAddress = excludedMember.getSocketAddress().toString();
+
+        Job streamJob = JobDefinition.createStreamJob(sleepBetweenStreamRecordsMs,
+                snapshotIntervalMs, client, excludedMemberUuid, name);
+        waitForJobStatus(streamJob, RUNNING);
+
+        long validationCount = 0;
+        while (System.currentTimeMillis() - begin < durationInMillis) {
+            if (getJobStatusWithRetry(streamJob) == JobStatus.FAILED) {
+                throw new RuntimeException(logPrefix + "Job failed: " + streamJob.getName());
+            }
+
+            sleepMinutes(sleepBeforeValidationInMinutes);
+
+            assertStreamSinkMap(client, excludedMemberAddress, name);
```

Comment:
I think race condition is still here - if server is restarted map will be empty and assertion will fail. We should configure persistence for the map. Or we can use remoteMap on stable cluster for isolated cluster (for this case we should also probably increase time for checking back to 2minutes - since sink can be cleared and check after 1minute but isolated cluster is down for 30seconds during the whole restart and there can be some other small delay which could mean the job is not started too fast). WDYT?

Reply:
@olukas just for sake of my curiosity - do you restart job or cluster during the soak test?

Reply:
@Fly-Style cluster is restarted during the soak tests (there are 2 clusters, one stable which is not restarted, and one dynamic which is restarted node by node and then whole cluster)